modules/registrars/openprovider/src/TldPriceCache.php

```diff
@@ -14,17 +15,43 @@ class TldPriceCache
      * Check if the file exists.
      * @return bool
      */
-    public function has()
+    public function has($params)
     {
-        if(!@is_file($this->getLocation()))
+        $filePath = $this->getLocation();
+        $timePeriodInMinutes = 1440; // 24 hours
+
+        if (file_exists($filePath)) {
```

Comment:
I think better to move these logic also to the downloadTldCache function

Reply:
This change has been added